pkg/vm/state.go

```diff
@@ -1,25 +1,75 @@
 package vm
 
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
 // State of the VM.
 type State uint
 
 // Available States.
 const (
-	noneState State = iota
-	haltState
+	noneState State = 0
+	haltState State = 1 << iota
 	faultState
 	breakState
 )
 
+func (s State) HasFlag(f State) bool {
+	return s&f != 0
+}
+
 func (s State) String() string {
-	switch s {
-	case haltState:
-		return "HALT"
-	case faultState:
-		return "FAULT"
-	case breakState:
-		return "BREAK"
-	default:
+	if s == noneState {
 		return "NONE"
 	}
+
+	ss := make([]string, 0, 3)
+	if s.HasFlag(haltState) {
+		ss = append(ss, "HALT")
+	}
+	if s.HasFlag(faultState) {
+		ss = append(ss, "FAULT")
+	}
+	if s.HasFlag(breakState) {
+		ss = append(ss, "BREAK")
+	}
+	return strings.Join(ss, ", ")
+}
+
+func StateFromString(s string) (st State, err error) {
+	if s = strings.TrimSpace(s); s == "NONE" {
+		return noneState, nil
+	}
+
+	ss := strings.Split(s, ",")
+	for _, state := range ss {
+		switch state = strings.TrimSpace(state); state {
+		case "HALT":
+			st |= haltState
+		case "FAULT":
+			st |= faultState
+		case "BREAK":
+			st |= breakState
+		default:
+			return 0, errors.New("unknown state")
```

Comment:
0 == NONE, propose to use 255

Reply:
We return error. This is not a state but a mask and in case of error nothing should be applied to it (which is the meaning of "none").